### PR TITLE
Fixed #29058 -- return geom type for GeoDjango fields when using Spatialite backend so that migrations don't fail.

### PR DESCRIPTION
**Pull request by v1kku**
[Ticket 29058](https://code.djangoproject.com/ticket/29058)

### CODE REVIEW

File: docs/releases/2.2.8.txt [hunk 1]
```diff
@@ -10,4 +10,4 @@ Django 2.2.8 fixes several bugs in 2.2.7 and adds compatibility with Python
 Bugfixes
 ========
 
-* ...
+* Fixed migration crash when a GeoDjango model field is altered when using Spatialite backend. (:ticket:`29058`)
```
Comment (felixxm) [inline docs/releases/2.2.8.txt]: This patch doesn't qualify for a backport, please remove release notes.

Comment (v1kku) [reply docs/releases/2.2.8.txt]: Done

**Comment by felixxm** [comment]
@jtiai Jani, Can you take a look?

**Comment by felixxm** [comment]
@sir-sigurd Can you take a look?

**Comment by v1kku** [comment]
@sir-sigurd  This change is **not** meant to silence those failures, sorry if the title is misleading. Currently you cannot run any AlterField operation during migrations on any GeoDjango fields when using the spatialite backend, including adding a `verbose_name` or renaming the field. Any failures from the backend or database itself will still raise an exception, this change doesn't catch those.

**Comment by sir-sigurd** [comment]
> This change is **not** meant to silence those failures

It doesn't look like this is tested somewhere, i.e. you should unskip the mentioned `test_alter_geom_field_dim` and assert that `self.alter_gis_model` still raises exception.

**Comment by v1kku** [comment]
> > This change is **not** meant to silence those failures
> 
> It doesn't look like this is tested somewhere

I don't think that's within the scope of this PR. It would be testing behaviour that this change doesn't touch on at all. This change does not affect error handling, it is not silencing an error, it is fixing an old problem that was causing an error. So testing that db errors are raised would not be testing the changes made here. Perhaps that needs a new ticket? Again, sorry for the misleading title, a better one is "return geom type for GeoDjango fields when using Spatialite backend to allow AlterField migrations to run"

> i.e. you should unskip the mentioned test_alter_geom_field_dim

That test is testing a field that is [only supported by postgis](https://docs.djangoproject.com/en/2.2/ref/contrib/gis/model-api/#dim), the decorator should actually be `@skipUnless(postgis)` because that test is only supposed to pass on postgis.

**Comment by nessita** [comment]
@v1kku Hi! I'm doing some PR cleanup, would you have time to update this PR with the latest main resolving the reported conflict? Then we can re-ping Sergey regarding your latest comments. Thank you!

**Comment by nessita** [comment]
Closing due to inactivity, feel free to re-open if you can continue working on this. Thanks!

**Pull request closed without merging**